### ISSUE TITLE
return type for getContent()

### ISSUE THREAD

**Issue by tacman** [liip/LiipImagineBundle#1452]
The current implementation of getContent in FileBinary is
```php
    public function getContent(): string
    {
        return file_get_contents($this->path);
    }
```

But file_get_contents returns either a string or false.  So the return type will eventually need to be either string|false, or...

* throw an error
* return null
* return '' (typecast the result to string).

Not at all urgent, I was just playing around with what we'd see if we bumped to php 8.0 as a minimum and ran phpstan with level 7, and this showed up.

If null, we could do

```php
    public function getContent(): ?string
    {
         return ($content = file_get_contents($this->path)) === false ? null: $content;
    }
```

Of course, this requires changing the interface and everything that implements it.

Easiest is probably 
```php
    public function getContent(): string
    {
        return (string)file_get_contents($this->path);
    }
```

but this is returning a stringified false.

So maybe it's not worth doing anything.  Of course, 3.0 is the best time to make these kind of changes.

**Comment by dbu** [comment]
yeah with the annotations, it actually returns any of string|false. i think we should throw an exception rather than return false, it will make things much clearer.

tagging for version 3, and glad if you want to work on this. for version 2, i will fix the annotation to reflect reality - we can't change it there without a BC break.

**Comment by dbu** [comment]
sorry, scratch the part i wrote about streams - too early in the morning. file_get_contents never gives a stream

**Comment by dbu** [comment]
fixed in #1460
